File: exercism/javascript/acronym/acronym.js

const isLower = (char) => char.match(/[a-z]/)
const isUpper = (char) => char.match(/[A-Z]/)
const isAlpha = (char) => isLower(char) || isUpper(char)

const acronym = ({ res, prior }, char) => {
  if (prior === null ||
      !isAlpha(prior) && isAlpha(char) ||
      isLower(prior) && isUpper(char)) {
    res = res.concat(char.toUpperCase())
  }
  return { res, prior: char }
}

const parse = (phrase) =>
  phrase.split('').reduce(acronym, { res: '', prior: null }).res

module.exports = { parse }

File: exercism/csharp/acronym/Acronym.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Acronym
{
    public static string Abbreviate (string phrase)
    {
        var acronym = new List<char>();
        for (var i = 0; i < phrase.Length; i++) {
            var ch = phrase[i];
            var prior = i == 0 ? ' ' : phrase[i - 1];
            if (prior == ' ' ||
                prior == '-' ||
                Char.IsLower(prior) && Char.IsUpper(ch))
            {
                acronym.Add(Char.ToUpper(ch));
            }
        }
        return String.Concat(acronym);
    }

}
